src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+import React, { useState, useContext, useRef, useEffect } from 'react';
+import { Animated, TouchableOpacity } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import PropTypes from 'prop-types';
+
+import { makePxValue } from '../../_helpers/theme';
+import automation from '../../_helpers/automation-attributes';
+import View from '../View/';
+import isPropDefined from '../../_helpers/isPropDefined';
+
+const styles = {
+  size({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return {
+          containerSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.xxxlarge,
+          },
+          knobSize: {
+            height: theme.spacings.medium,
+            width: theme.spacings.medium,
+            activeWidth: theme.spacings.large,
+          },
+        };
+      case 'large':
+        return {
+          containerSize: {
+            height: makePxValue(3),
+            width: makePxValue(6),
+          },
+          knobSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.large,
+            activeWidth: theme.spacings.xxlarge,
+          },
+        };
+    }
+  },
+  padding({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return theme.spacings.xxsmall;
+      case 'large':
+        return theme.spacings.xsmall;
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+};
+
+const StyledContainer = styled(TouchableOpacity)(
+  (props) => `
+  height: ${styles.size(props).containerSize.height};
+  width: ${styles.size(props).containerSize.width};
+  border-radius: ${`${parseInt(styles.size(props).containerSize.height, 10) / 2}px`};
+  padding: ${styles.padding(props)};
+  flex-direction: row;
+  align-self: ${styles.alignSelf(props)};
+`,
+);
+
+const StyledKnob = styled(View)(
+  (props) => `
+  position: absolute;
+  align-self: center;
+  height: ${styles.size(props).knobSize.height};
+  border-radius: ${`${parseInt(styles.size(props).knobSize.height, 10) / 2}px`};
+  background-color: #fff;
+`,
+);
+
+const AnimatedContainer = Animated.createAnimatedComponent(StyledContainer);
+const AnimatedKnob = Animated.createAnimatedComponent(StyledKnob);
```

Comment:
New line above this please

Reply:
why here? it is just like declaring variables in continuous lines, like this:
```
const [toggleState, setToggleState] = useState(false);
const [active, setActive] = useState(false);
```